### DISCUSSION TITLE
Welcome to PowerQueryNet Discussions!

### DISCUSSION THREAD

**Discussion by gsimardnet** [gsimardnet/PowerQueryNet#14] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JaviOrdi** [comment]
Hi Guillaume,

First of all, thanks for your contribution to the development of new options to use M language outside Power Query. I am trying to use your code to my own developments and debugging to understand it. I have an issue with task Query Execution Result. When I run it synchronously I get an exception ("cannot create process"). I think mash up is well conformed (image below) but I can't get it run. PowerQueryNet Service is running on my machine. Can you give me any ideas?

Thanks in advance.

![image](https://user-images.githubusercontent.com/14080247/105471778-65a60f00-5c9b-11eb-8030-f642120370b5.png)
![image](https://user-images.githubusercontent.com/14080247/105472218-f0870980-5c9b-11eb-8b9b-457445fd0d83.png)

**Reply by gsimardnet** [reply]
Hello,

Do you have the same error if you simplify your query to the following expression ?

`let  
    Source = Text.Proper("Hello, World!")  
in  
    Source `

**Reply by JaviOrdi** [reply]
It worked! Thanks.

**Comment by Gravanita** [comment]
Hi Guillaume,
Can you give an example on how to invoke the DateTable2018.pq ?

**Reply by gsimardnet** [reply]
DateTable2018 depends on a user defined function fnDateTable.
So assuming that both DateTable2018.pq and fnDateTable.pq are in the ".\MyQueries" folder, the following PQNet command would work:
`pqnet ".\MyQueries" DateTable2018`

**Reply by Gravanita** [reply]
Thanks for your prompt reply. Its working.

**Comment by Gravanita** [comment]
Hi Guillaume,
I have M code that fetches information from an xlsx file.
When I try to invoke pqnet from the command line I get:
"Credentials are required to connect to the file source".

Its a local file. I don't understand the credentials it is expecting.

**Reply by gsimardnet** [reply]
Hi,

You need to create a credential file (e.g. #credentials.xml) with the content similar to this :
`<Credentials xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">`
`  <Credential xsi:type="CredentialFile">`
`  	<Path>C:\Temp\MyFile.xlsx</Path>`
`  </Credential>`
`</Credentials>`

Then you can use PQNet like the following :
`pqnet MyQuery.pq -c #credentials.xml`

**Reply by Gravanita** [reply]
Thanks, it is working. Is there anyway to pass in parameters?

**Reply by Gravanita** [reply]
If in the file path with have accent characters like "ã ç à" apparentely it does not work. How can I use these characters in the xml path element?

**Reply by gsimardnet** [reply]
I have tested and was able to reproduce your error. Unfortunately, this error is in the [Power Query SDK](https://marketplace.visualstudio.com/items?itemName=Dakahn.PowerQuerySDK). I suggest that you ask your question in their forum.

**Reply by jrb2971** [reply]
So login credentials are in the credential file? the credential file has to point to an excel file? I don't understand the column format of the excel file... shoudln't it be like user@domain.com (or some userid) and password?  I'm trying to get pqnet to work with Azure Insights (service fabric). Thanks in advance.

**Comment by NickEPalmer** [comment]
Hi,
We have been experimenting with PowerQueryNet and the Power Query M formula language and have found it amazingly powerful.  It appears from reading posts here and searching the net that the Power Query M formula APIs aren't something that Microsoft has made public yet.  Do you have any insight into if that's something they will be doing?
Thanks

**Reply by gsimardnet** [reply]
When I first started this project, I had some communications with Microsoft and they mentioned it was something they wish to do but I haven't got any feedback since then.

**Reply by NickEPalmer** [reply]
Ok thanks for the information.  We haven't reached out to Microsoft regarding this but seems like something we should do. If I wanted to contact you outside of here, is there anyway to do that?

**Reply by gsimardnet** [reply]
You can reach me on my [LinkedIn](https://ca.linkedin.com/in/gsimard)

**Reply by NickEPalmer** [reply]
Hi I sent you a message on LinkedIn - did you get it?

**Reply by gsimardnet** [reply]
Yes, I just responded. Thanks

**Comment by Adhithya93** [comment]
Hi Can you help me with build issue I am facing.
Tried to build the master branch using visual studio 2019.
-The dlls under Engine/lib are not available
![Uploading Capture.JPG…]()

**Reply by gsimardnet** [reply]
If you have installed PowerQueryNet, you can find them under C:\Program Files\PowerQueryNet.
Otherwise, you can download [PowerQuerySdk.vsix 1.0.0.16](http://dakahn.gallery.vsassets.io/_apis/public/gallery/publisher/dakahn/extension/powerquerysdk/1.0.0.16/assetbyname/PowerQuerySdk.vsix) and unzip this package to find the required libraries.
Then copy those files in the output directory.

Sorry for the inconvenience, I cannot include these files in my project because I don't have the right to distribute them.

**Comment by edurod18** [comment]
Hey, I am having trouble getting this to run:

![image](https://github.com/gsimardnet/PowerQueryNet/assets/7401259/811ace6f-0217-46f3-adbc-6a0277f9c3c2)


I get this exception:
"Method not found: 'System.Threading.Tasks.Task`1<Microsoft.Mashup.Tools.QueryExecutionResults> Microsoft.Mashup.Tools.QueryExecutor.CreateExecution(Microsoft.Data.Mashup.MashupConnectionStringBuilder, Microsoft.Mashup.Tools.CredentialStore, Boolean)'."

Any ideas what I am missing.
